takwimu/dataset_release_utils.py

```diff
@@ -0,0 +1,51 @@
+from django.conf import settings
+
+
+def get_country_geo_code(geo):
+    level = geo.geo_level.lower()
+    country = geo.geo_code \
+        if level == 'country' \
+        else geo.ancestors()[-1].geo_code
+
+    return country
+
+
+def get_page_releases_per_country(dataset_name, geo, year, filter_releases=True):
+    """
+    Return the active release being viewed and a list of related releases
+    for a geo and dataset
+    """
+    from wazimap.models import Dataset
+
+    releases = {}
+    releases.setdefault('other', [])
+
+    query = Dataset.objects.get(name=dataset_name).releases.order_by('-year')
+
+    # get available release years for this particular level
+    available_years = settings.HURUMAP['available_release_years'].get(
+        geo.geo_level, 9000)
+
+    # get the available releases for this country
+    country_code = get_country_geo_code(geo)
+    print '\n\n\n\n\n\n\n\n\n\n'
+    print country_code
+    print '\n\n\n\n\n\n\n\n\n\n'
+    country_available_years = settings.HURUMAP[
+        'releases_available_per_country'].get(country_code, 8000)
+    
+    # get the available years for the level and country in which chosen geography exists
+    available_years = list(set(available_years).intersection(set(country_available_years)))
```

Comment:
Just thinking out loud here, is this necessary? If we already have `country_available_years`, why don't we just return that? Because if we **don't** have `country_available_years`, then the intersection will always be empty.

Reply:
@kilemensi my thoughts were; if let's say the available releases at the level1 are `[2014]` but TZ only has releases for `[2014, 2017]` so instead of returning all the country, so the intersection would essentially the available for that country at that geography level

Reply:
Then I think we misunderstood each other @phillipahereza.

My thinking is `releases_available_per_country` will now hold the same structure as `available_release_years` except it will be per country. i.e.
```python
HURUMAP['releases_available_per_country'] = {
    'UG': {
        'country': [2017]
    }
}
```
`available_release_years` only contains `default` value to make Wazimap happy.